src/deferred/exceptionHook.js

```diff
@@ -0,0 +1,21 @@
+define( [
+	"../core",
+	"../deferred"
+], function( jQuery ) {
+
+// These usually indicate a programmer mistake during development,
+// warn about them ASAP rather than swallowing them by default.
+var rerrorNames = /^(Eval|Internal|Range|Reference|Syntax|Type|URI)Error$/;
+
+jQuery.Deferred.exceptionHook = function( error, stack ) {
+
+	// Support: IE9
+	// Console exists when dev tools are open, which can happen at any time
+	var warn = window.console && window.console.warn;
+
+	if ( warn && error && rerrorNames.test( error.name ) ) {
+		warn.call( window.console, "jQuery.Deferred exception: " + error.message, stack );
```

Comment:
Yeah, technically I could just let gzip do its thang and avoid the variable, which would also remove the need for the `.call`. I'm okay with that if people prefer it.